rust_src/src/process.rs

```diff
@@ -84,15 +89,52 @@ impl LispProcessRef {
     }
 }
 
+/// This is how commands for the user decode process arguments.  It
+/// accepts a process, a process name, a buffer, a buffer name, or nil.
+/// Buffers denote the first process in the buffer, and nil denotes the
+/// current buffer.
+pub fn get_process(name: LispObject) -> LispProcessRef {
+    let proc_or_buf = name.is_string() {
+        let process = lisp_get_process(name);
+
+        if process.is_nil() {
+            let buffer = get_buffer(name);
+
+            if buffer.is_nil() {
+                panic!("Process {} does not exist", name.as_string_or_error())
+            } else {
+                buffer
+            }
+        } else {
+            process
+        }
+    } else if name.is_nil() {
+        // check if buffer is dead
+        get_buffer(current_buffer())
+    } else {
+        name
+    }
+
+    if proc_or_buf.is_buffer() {
+        let proc = get_buffer_process(proc_or_buf);
+        if proc.is_nil() {
+            panic!("Buffer {} has no process", name.as_string_or_error())
+        }
+        proc
+    } else {
+        // check process
```

Comment:
Should this not return the process?

Reply:
There's some stuff in the original C function that does some extra validation (there are bits and pieces to see if the process is actually running or not, it seems). I just need to figure out what exactly that is.